lib/collection/src/collection/snapshots.rs

```diff
@@ -135,23 +134,10 @@ impl Collection {
         let _temp_path = TempPath::from_path(&snapshot_path_tmp_move);
         fs::copy(&snapshot_temp_arc_file.path(), &snapshot_path_tmp_move).await?;
 
-        // compute and store the file's checksum before the final snapshot file is saved
-        // to avoid making snapshot available without checksum
-        let checksum_path = snapshot_ops::get_checksum_path(&snapshot_path);
-        let checksum = hash_file(&snapshot_path_tmp_move).await?;
-        let checksum_file = tempfile::TempPath::from_path(&checksum_path);
-        let mut file = tokio::fs::File::create(checksum_path.as_path()).await?;
-        file.write_all(checksum.as_bytes()).await?;
-
-        let snapshot_file = tempfile::TempPath::from_path(&snapshot_path);
-        fs::rename(&snapshot_path_tmp_move, &snapshot_path).await?;
-
-        // Snapshot files are ready now, so keep them
-        snapshot_file.keep()?;
-        checksum_file.keep()?;
-
-        log::info!("Collection snapshot {snapshot_name} completed into {snapshot_path:?}");
-        snapshot_ops::get_snapshot_description(&snapshot_path).await
+        let snapshot_storage_manager = &self.snapshot_storage_manager;
+        snapshot_storage_manager
+            .store_file(&snapshot_path_tmp_move, snapshot_path.as_path())
```

Comment:
Actual moving to another location (mount or S3) happens n the `fs::copy` statement above. 
Therefore it should also be habdled by `snapshot_storage_manager`

Reply:
Not related to this PR, but checksum should be computed before copying

Reply:
Right. I moved it, also moved the checksum computation before copying.